util/src/test/java/com/google/cloud/hadoop/util/RetryHttpInitializerTest.java

```diff
@@ -166,4 +248,22 @@ private static RetryHttpInitializer createRetryHttpInitializer(Credentials crede
             .setReadTimeout(Duration.ofSeconds(5))
             .build());
   }
+
+  private class TestRetryHttpInitializer extends RetryHttpInitializer {
+    private boolean init;
+
+    public TestRetryHttpInitializer(Credentials credentials, RetryHttpInitializerOptions build) {
+      super(credentials, build);
+    }
+
+    @Override
+    protected RequestTracker getRequestTracker(HttpRequest request) {
```

Comment:
Correct me If I am wrong, With this override what we are able to use the same RequestTracker instance against all the gcs requests made in test case i.e. getMetadata, getObject, listObject etc, all the retires of the request are also captured via same tracer but in actual production scenario it will not be the case, every request will have a separate tracker.

If yes, will it even be a valid test then?

Reply:
Yeah, same tracker will be used across all retries of same request, but if there are multiple requests, say two different request to access object metadata of object1 and object2. `TestRetryHttpInitializer` configured here will not be able to provide different tracker but provide the same tracker which will yield weird outcomes at verification step.

Reply:
will sync offline.